app/src/main/java/com/kamron/pogoiv/OcrHelper.java

```diff
@@ -486,14 +486,14 @@ private String getCandyNameFromImg(Bitmap pokemonImage) {
      * @param pokemonImage the image of the whole pokemon screen
      * @return a string which should remain the same even if you power up a pokemon
      */
-    private String getPokemonIdentifierFromImg(Bitmap pokemonImage) {
+    private Optional<String> getPokemonIdentifierFromImg(Bitmap pokemonImage) {
         Bitmap infoRow = Bitmap.createBitmap(pokemonImage,
                 (int)Math.round(widthPixels * .1f), (int)Math.round(heightPixels / 1.714286f),
                 (int)Math.round(widthPixels * .8f), (int)Math.round(heightPixels / 25.26316f));
         tesseract.setImage(infoRow);
         String uniqueText = tesseract.getUTF8Text();
 
-        return uniqueText;
+        return Optional.fromNullable(uniqueText);
```

Comment:
You are right we get empty sting for empty ocr. Optional is not better but its consistant with the rest of the code. Either way is fine with me.